src/engine/GroupBy.h

```diff
@@ -157,49 +157,68 @@ class GroupBy : public Operation {
   // `?z`.
   bool computeGroupByForJoinWithFullScan(IdTable* result);
 
-  // TODO: Update documentation
+  // Create a HashMap containing the results of each aggregation for each group.
   template <size_t OUT_WIDTH, size_t numAggregates>
-  bool computeGroupByForSortedSubtree(IdTable* result,
-                                      std::vector<Aggregate> aggregates,
-                                      const IdTable& subresult,
-                                      size_t columnIndex,
-                                      LocalVocab& localVocab);
-
-  struct ExplicitlySortedData {
+  bool computeGroupByForHashMapOptimization(IdTable* result,
+                                            std::vector<Aggregate> aggregates,
+                                            const IdTable& subresult,
+                                            size_t columnIndex,
+                                            LocalVocab* localVocab);
+
+  // Required data to perform HashMap optimization.
+  struct HashMapOptimizationData {
     size_t subtreeColumnIndex_;
   };
 
+  // Data to perform the AVG aggregation using the HashMap optimization.
   struct AverageAggregationData {
-    bool _error = false;
-    double _sum = 0;
-    int64_t _count = 0;
+    bool error_ = false;
+    double sum_ = 0;
+    int64_t count_ = 0;
     void increment(sparqlExpression::detail::NumericValue intermediate) {
       if (const int64_t* intval = std::get_if<int64_t>(&intermediate))
-        _sum += (double)*intval;
+        sum_ += (double)*intval;
       else if (const double* dval = std::get_if<double>(&intermediate))
-        _sum += *dval;
+        sum_ += *dval;
       else
-        _error = true;
-      _count++;
+        error_ = true;
+      count_++;
     };
     [[nodiscard]] ValueId calculateResult() const {
-      if (_error)
+      if (error_)
         return ValueId::makeUndefined();
       else
-        return ValueId::makeFromDouble(_sum / (double)_count);
+        return ValueId::makeFromDouble(sum_ / (double)count_);
```

Comment:
all c-style casts  like `(double)` are fishy/potentially unsafe in c++ and should be replace by 
`static_cast<double)(whatever)`. 
Doesn't sonarcloud complain there?